app/src/main/java/com/kks/nimblesurveyjetpackcompose/viewmodel/home/HomeViewModel.kt

```diff
@@ -1,75 +1,89 @@
 package com.kks.nimblesurveyjetpackcompose.viewmodel.home
 
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.kks.nimblesurveyjetpackcompose.di.IoDispatcher
 import com.kks.nimblesurveyjetpackcompose.model.ErrorModel
 import com.kks.nimblesurveyjetpackcompose.model.ResourceState
 import com.kks.nimblesurveyjetpackcompose.model.Survey
 import com.kks.nimblesurveyjetpackcompose.repo.home.HomeRepo
+import com.kks.nimblesurveyjetpackcompose.util.extensions.ErrorType
 import com.kks.nimblesurveyjetpackcompose.util.extensions.mapError
 import dagger.hilt.android.lifecycle.HiltViewModel
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.StateFlow
 import kotlinx.coroutines.flow.asStateFlow
+import kotlinx.coroutines.flow.update
 import kotlinx.coroutines.launch
 import javax.inject.Inject
 
 const val DEFAULT_PAGE_SIZE = 5
 const val DEFAULT_PAGES = 1
 private const val START_SURVEY_NUMBER = 0
 
+data class HomeUiState(
+    val surveyList: List<Survey> = emptyList(),
+    val userAvatar: String? = null,
+    val error: ErrorModel? = null,
+    val isRefreshing: Boolean = false,
+    val selectedSurveyNumber: Int = START_SURVEY_NUMBER
+)
+
+class HomeUiStatePreviewParameterProvider : PreviewParameterProvider<HomeUiState> {
+    override val values = sequenceOf(
+        HomeUiState(
+            surveyList = listOf(
+                Survey("", "", "Title1", "Description1"),
+                Survey("", "", "Title2", "Description2"),
+            )
+        ),
+        HomeUiState(
+            surveyList = listOf(
+                Survey("", "", "Title1", "Description1"),
+                Survey("", "", "Title2", "Description2"),
+            ),
+            error = ErrorModel(ErrorType.INFO, "Error")
+        ),
+        HomeUiState(error = ErrorModel(ErrorType.INFO, "Error")),
+        HomeUiState(isRefreshing = true)
+    )
+}
+
 @HiltViewModel
 class HomeViewModel @Inject constructor(
     private val homeRepo: HomeRepo,
     @IoDispatcher private val ioDispatcher: CoroutineDispatcher,
 ) : ViewModel() {
 
-    private var _error = MutableStateFlow<ErrorModel?>(null)
-    private val _isRefreshing = MutableStateFlow(false)
-    private val _surveyList = MutableStateFlow<List<Survey>>(listOf())
-    private val _userAvatar = MutableStateFlow<String?>(null)
     private var _pageCount = DEFAULT_PAGES
     private var _records = DEFAULT_PAGES
-    private val _selectedSurveyNumber = MutableStateFlow(START_SURVEY_NUMBER)
+    private val _homeUiState = MutableStateFlow(HomeUiState())
+
+    val homeUiState: StateFlow<HomeUiState>
```

Comment:
How about placing them together?
```suggestion
    private val _homeUiState = MutableStateFlow(HomeUiState())
    val homeUiState: StateFlow<HomeUiState>
```